### PR TITLE
Add teardown handling for appliance mode

### PR DESCRIPTION
**Pull request by suchita-g**
Signed-off-by: suchita.gatfane <sgatfane@redhat.com>

### CODE REVIEW

**Comment by suchita-g** [comment]
Destroy Job Testing : https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/397

**Comment by suchita-g** [comment]
Deployment JOB: Confirmation of creation of metadat.json at cluster path after appliance mode provider cluster creation
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/398/

**Review by dahorak — DISMISSED** [review]

File: ocs_ci/utility/rosa.py [hunk 1]
```diff
@@ -491,6 +499,58 @@ def delete_operator_roles(cluster_id):
...
+        status = cluster_info["status"]["state"]
+        logger.info(f"Cluster uninstalling status: {status}")
+        if status == "uninstalling":
+            logger.info(f"Cluster '{cluster}' is uninstalling")
+            break
+    for service_status in utils.TimeoutSampler(
+        1000, 30, get_rosa_service_details, cluster
+    ):
+        if "deleting service" in service_status:
+            logger.info(" Rosa service status is 'deleting service'")
```
Comment (vavuthu) [inline ocs_ci/utility/rosa.py:550]:
```suggestion
            logger.info("Rosa service status is 'deleting service'")
```

**Comment by suchita-g** [comment]
Verification deployment + Teardown Job using multicluster pipeline : https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/405/

**Comment by suchita-g** [comment]
Metadata.json creation at cluster path verified: 
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/14005/
`2022-06-28 11:13:54  05:43:52 - MainThread - /home/jenkins/workspace/qe-deploy-ocs-cluster/ocs-ci/ocs_ci/utility/rosa.py - INFO  - service sgatfane-27jp5 is ready
2022-06-28 11:13:54  05:43:52 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ocm describe cluster sgatfane-27jp5 --json=true`

**Review by vavuthu — DISMISSED** [review]

**Comment by suchita-g** [comment]
Verified deployment and teardown with multicluster Job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/400/

**Review by vavuthu — APPROVED** [review]

**Review by jilju — APPROVED** [review]

**Pull request merged**